CS483/CS483/Kartaclysm/Components/ComponentKartStats.cpp

```diff
@@ -0,0 +1,90 @@
+//----------------------------------------------------------------------------
+// ComponentKartStats
+// Author: David MacIntosh
+//
+// Component for the base stats of a kart
+//----------------------------------------------------------------------------
+
+#include "ComponentKartStats.h"
+
+#include "ComponentKartController.h"
+
+namespace Kartaclysm
+{
+	ComponentKartStats::ComponentKartStats(
+		HeatStroke::GameObject* p_pGameObject,
+		int p_iMaxSpeed,
+		int p_iAcceleration,
+		int p_iHandling,
+		int p_iDurability)
+		:
+		Component(p_pGameObject),
+		m_iMaxSpeedStat(p_iMaxSpeed),
+		m_iAccelerationStat(p_iAcceleration),
+		m_iHandlingStat(p_iHandling),
+		m_iDurabilityStat(p_iDurability)
+	{
+		ComponentKartController* kartController = (ComponentKartController*)m_pGameObject->GetParent()->GetComponent("GOC_KartController");
+
+		kartController->SetKartStats(m_iMaxSpeedStat, m_iAccelerationStat, m_iHandlingStat, m_iDurabilityStat);
```

Comment:
If the creation of these stats rely on KartController, how do we access them in the kart/driver selection screen? Is that your task or David Hanna's?